SECURITY.WHITEPAPER.md

```diff
@@ -0,0 +1,139 @@
+# Datasafe Security Whitepaper
+
+## General information
+Datasafe is a flexible encryption library. It uses different encryption algorithms. Some of them can be 
+configured by client application. Under the hood Datasafe uses BouncyCastle library to perform encryption.
+
+CMS (Cryptographic Message Syntax) standard [RFC5652](https://tools.ietf.org/html/rfc5652.html) employed for storing private 
+files encrypted with symmetric keys as well as for sharing files with other users using asymmetric key pairs. 
+
+## Default locations
+To maintain information where user data resides, Datasafe uses concept of user profile. There are two types of user 
+profile: private and public
+
+<details>
+  <summary>Default location within system dfs</summary>
+  
+    ```
+        /profiles
+            /private
+                /username - user's private profile 
+            /public
+                /username - user's public profile   
+        /users
+            /public
+                /pubkeys - keystore consists user's public key
+                /inbox - location of shared with user files    
+            /private
+                /keystore - keystore consists user's private key
+                /files/SIV - location of private files. SIV is 3 symbol path encryption algorithm identifier.
+    ```
+</details>
+      
+<details>             
+  <summary>Example private profile:</summary>
+  
+```
+    {
+        "keystore": {
+            "resource": "s3://bucketname/users/username/private/keystore"
+        },
+        "privateStorage": [
+            [{"id": "DEFAULT"}, {"resource": "s3://bucketname/users/username/private/files/"}]
+        ],
+        "inboxWithFullAccess": {
+            "resource": "s3://bucketname/users/username/public/inbox/"
+        },
+        "publishPublicKeysTo": {
+            "resource": "s3://bucketname/users/username/public/pubkeys"
+        },
+        "associatedResources": [
+            {"resource": "s3://bucketname/users/username/"}
+        ],
+        "documentVersionStorage": {
+            "resource": "s3://bucketname/users/username/versions/"
+        },
+        "appVersion": "BASELINE"
+    }
+```
+</details>
+         
+<details>   
+  <summary>Example public profile:</summary>
+  
+```
+    {
+        "publicKeys": {
+            "resource": "s3://bucketname/users/username/public/pubkeys"
+        },
+        "inbox": {
+            "resource": "s3://bucketname/users/username/public/inbox/"
+        },
+        "appVersion": "BASELINE"
+    }   
+```
+</details>
+
+## Keystore encryption
+System wide password is used to open keystore and users' personal password to read users' keys. Password can be changed 
+without the need of changing keystore content.
+By default used:
```

Comment:
Also it is good to add link to EncryptionConfig class that provides these defaults